arbos/programs/programs.go

```diff
@@ -24,15 +25,18 @@ type Programs struct {
 	programs       *storage.Storage
 	moduleHashes   *storage.Storage
 	dataPricer     *DataPricer
+	cacheManagers  *addressSet.AddressSet
 }
 
 type Program struct {
 	version       uint16
-	initGas       uint24
-	asmEstimateKb uint24 // Predicted size of the asm
+	initGas       uint16
+	cachedInitGas uint16
```

Comment:
do we really want initGas + cachedInitGas per program?
it both complicates code and makes some changes harder.
I think we only want an initGas per contract, and a general param(s?) for discount. That'll give owner an easy handle that changes how much discount is given for being in cache (which it'll make sense to change while changing cache policy - you can give a larger discount if you cache 16 contracts vs if you cache 1024 of them)

Reply:
The reason for having both `initGas` and `cachedInitGas` is that programs vary in their initialization timings. I ran some benchmarks and found that it's not as simple as "cached is 8x cheaper" or "uncached pays an extra Y gas". So basically the variability of the savings motivates recording both during program activation.